packages/client/hmi-client/src/components/models/tera-model-type-legend.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+	<section class="legend" v-if="stateTypes || transitionTypes">
+		<ul>
+			<li v-for="(type, i) in stateTypes" :key="i">
+				<div class="legend-key-circle" :style="getLegendKeyStyle(type ?? '')" />
+				{{ type }}
+			</li>
+		</ul>
+		<ul>
+			<li v-for="(type, i) in transitionTypes" :key="i">
+				<div class="legend-key-square" :style="getLegendKeyStyle(type ?? '')" />
+				{{ type }}
```

Comment:
I don't understand why you need to do `type ?? ''` , what are you guarding against? why would it be undefined? Can we drop it in favour of cleaner intention/syntax?